### PR TITLE
2018-07 @rudeboybert edits

### PR DESCRIPTION
**Pull request by rudeboybert**
Even though I'm starting the pull request now, @ismayc no action needed until I formally request review when I'm done updating all Chapters.

### CODE REVIEW

**Comment by rudeboybert** [comment]
Again, a review on your part @ismayc would be nice, but is not essential at this point. I still need to incorporate TODO's for Ch7 & 8, which I'll do in a new PR.

File: 03-visualization.Rmd [hunk 1]
```diff
@@ -673,9 +719,16 @@ ggplot(data = weather, mapping = aes(x = factor(month), y = temp)) +
   geom_boxplot()
 ```
 
-We have introduced a new function called `factor()` here.  One of the things this function does is to convert a discrete value like `month` (1, 2, ..., 12) into a categorical variable.  The "box" part of this plot represents the 25^th^ percentile, the median (50^th^ percentile), and the 75^th^ percentile.  The dots correspond to *outliers*.  (The specific formulation for these outliers is discussed in Appendix \@ref(appendixA).)  The lines show how the data varies that is not in the center 50% defined by the first and third quantiles.  Longer lines correspond to more variability and shorter lines correspond to less variability.
+We have introduced a new function called `factor()` here which converts a discrete numerical values, like `month` whose values are the numbers 1 through 12, into a categorical variable variable with "labels" 1 through 12. So for example, after converting `month` to a categorical variable, we couldn't perform any mathematical operations on its values. Note also that the labels have an ordering on the x-axis in Figure \@ref(fig:monthtempbox); this ordering is based on the alphanumeric ordering of the numbers 1 through 12.
```
Comment (ismayc) [inline 03-visualization.Rmd]: Not sure this wording makes more sense. It seems a little too wordy and run-on? Let's revisit cleaning it up together.

Comment (rudeboybert) [reply 03-visualization.Rmd]: Feel free to break up my run-ons. I do tend to favor run-ons the first time I write something, but then break them up in the polishing phase

File: 04-tidy.Rmd [hunk 2]
```diff
@@ -326,31 +384,40 @@ We'll see in Chapter \@ref(wrangling) how we could use the `mutate()` function t
...
+```
+```{r lc4-3solutions-4, include=show_solutions('4-3')}
+dem_score
+dem_score_tidy
+```
+```{asis lc4-3solutions-5, include=show_solutions('4-3')}
+**`r paste0("(LC", chap, ".", (lc - 1), ")")`** The code is similar
+```
+```{r lc4-3solutions-6, include=show_solutions('4-3'), echo=show_solutions('4-3'), message=FALSE, warning=FALSE}
+life_expectancy <- read_csv('http://ismayc.github.io/le_mess.csv')
```
Comment (ismayc) [inline 04-tidy.Rmd]: We'll have to make sure to remember to move these data sets all over to moderndive.com

File: 05-wrangling.Rmd [hunk 3]
```diff
@@ -65,9 +67,24 @@ library(nycflights13)
 ```{r message=FALSE, warning=FALSE, echo=FALSE, purl=FALSE}
 # Packages needed internally, but not in text.
 library(knitr)
+library(readr)
 ```
 
 
+### DataCamp {-}
+
+Our approach to introducing data wrangling tools from the `dplyr` package is very similar to the approach taken in [David Robinson's](https://twitter.com/drob) DataCamp course "Introduction to the Tidyverse," a course targetted at people new to R and the tidyverse. If you're interested in complementing your learning below in an interactive online environment, click on the image below to access the course. The relevant chapters are Chapter 1 on "Data wrangling" and Chapter 3 on "Grouping and summarizing".
```
Comment (ismayc) [inline 05-wrangling.Rmd]: "targetted" should be "targeted"

File: 05-wrangling.Rmd [hunk 4]
```diff
@@ -236,7 +253,7 @@ summary_temp <- weather %>%
 kable(summary_temp)
 ```
 
-It is not good practice to include a `na.rm = TRUE` in your summary commands by default; you should attempt to run them without this argument.  This is because removing missing data can have an impact on your analyses. In fact, an entire branch of the field of statistics deals with missing data. The take away point is that `na.rm = TRUE` should only be used after you become aware of the implications of its use (see the Learning Checks below for an example).
+It is not good practice to include a `na.rm = TRUE` in your summary commands by default; you should attempt to run code first without this argument as this will alert you to the presense of missing data. Only after you've identified where missing values occur and have thought about the potential causes of this missing should you consider using `na.rm = TRUE`. In the upcoming Learning Checks we'll consider the possible ramifications of blindly sweeping rows with missing values under the rug.
```
Comment (ismayc) [inline 05-wrangling.Rmd]: "presence" instead of "presense"

File: 06-regression.Rmd [hunk 5]
```diff
@@ -203,66 +207,113 @@ ggplot(data = values, mapping = aes(V1, V2)) +
   )
 ```
 
-The correlation coefficient is computed using the `cor()` function, where in this case the inputs to the function are the two numerical variables from which we want to calculate the correlation coefficient. Recall from Subsection \@ref(exploredataframes) that the `$` pulls out specific variables from a data frame:
+The correlation coefficient is computed using the `get_correlation()` function in the `moderndive` package, where in this case the inputs to the function are the two numerical variables from which we want to calculate the correlation coefficient. We place the name of the response variable on the left hand side of the `~` and the explanatory variable on the right hand side of the "tilde." We will use this same "formula" syntax with regression later in this chapter.
+
```
Comment (ismayc) [inline 06-regression.Rmd]: We can use `moderndive::get_correlation()` here now.

File: 06-regression.Rmd [hunk 6]
```diff
@@ -203,66 +207,113 @@ ggplot(data = values, mapping = aes(V1, V2)) +
...
+  labs(x = "Beauty Score", y = "Teaching Score", 
+       title = "Jittered scatterplot") +
+  geom_path(data = box, aes(x=x, y=y), col = "orange", size = 1)
+p1 + p2
 ```
 
-From Figure \@ref(fig:numxplot2) we make several observations:
+We make several further observations:
 
 <!-- We might want to actually highlight these points in the plot. -->
```
Comment (ismayc) [inline 06-regression.Rmd]: We could use {gghighlight} to do so I think.

File: 06-regression.Rmd [hunk 7]
```diff
@@ -587,7 +670,7 @@ gapminder2007 <- gapminder %>%
   select(country, continent, lifeExp, gdpPercap)
 ```
 
-Let's look at the raw data values both by bringing up RStudio's spreadsheet viewer and the `glimpse()` function, although in Table \@ref(tab:model2-data-preview) we only show 5 randomly selected countries out of `r nrow(gapminder2007)`:
+Let's you should look at the raw data values both by bringing up RStudio's spreadsheet viewer and the `glimpse()` function. In Table \@ref(tab:model2-data-preview) we only show 5 randomly selected countries out of `r nrow(gapminder2007)`:
```
Comment (ismayc) [inline 06-regression.Rmd]: Change to "You should" instead of "Let's you should"

File: 06-regression.Rmd [hunk 8]
```diff
@@ -755,25 +841,90 @@ get_regression_table(lifeExp_model) %>%
...
+
+$$
+\mathbb{1}_{\mbox{Amer}}(x) = \left\{
+\begin{array}{ll}
+1 & \text{if } \text{country } x \text{ is in the Americas} \\
+0 & \text{otherwise}\end{array}
+\right.
+$$
+
+Now let's intepret the terms in the estimate column of the regression table. First $b_0 =$ `intercept = 54.8` corresponds to the mean life expectancy for countries in Africa, since for country $x$ in Africa we have the following equation:
```
Comment (ismayc) [inline 06-regression.Rmd]: "interpret" instead of "intepret." I'll do a spellcheck here to make sure I'm not missing others.

File: 06-regression.Rmd [hunk 9]
```diff
@@ -943,14 +1098,41 @@ ggplot(data = values, mapping = aes(V1, V2)) +
...
+
+For example, let's say an instructor has their `bty_avg` reevaluated, but only after taking steps to try to boost their beauty score. Does this mean that they will suddenly be a better instructor? Or will they suddenly get higher teaching scores? Maybe? 
+
+Here is another example, a not-so-great medical doctor goes through their medical records and finds that patients who slept with their shoes on tended to wake up more with headaches. So this doctor declares "Sleeping with shoes on cause headaches!" 
+
+```{r moderndive-figure-causal-graph-2, echo=FALSE, fig.align='center', fig.cap="Does sleeping with shoes on cause headaches?"}
+knitr::include_graphics("images/flowcharts/flowchart.010-cropped.png")
+```
+
+However as some of you might have guessed, if someone is sleeping with their shoes on its probably because they are drunk. Furthermore, drinking more tends to cause more hangovers, and hence more headaches.
```
Comment (ismayc) [inline 06-regression.Rmd]: "intoxicated" instead of "drunk" maybe?

File: 07-multiple-regression.Rmd [hunk 10]
```diff
@@ -293,15 +304,17 @@ a) Looking at the raw values
 a) Computing summary statistics of the variables of interest.
 a) Creating informative visualizations
 
-What can you say about the relationship between a credit card holder's balance and their credit rating and age? Recall that if you've lost some information from the dataset in it's original form, you can reload a fresh copy using `data("Credit")`. 
+What can you say about the relationship between a credit card holder's balance and their credit rating and age? Recall that if you've lost some information from the dataset in its original form by using `select()` with `<-`, you can reload a fresh copy using `data("Credit")`.
```
Comment (ismayc) [inline 07-multiple-regression.Rmd]: Are we going to be removing this?

File: 12-thinking-with-data.Rmd [hunk 11]
```diff
@@ -2,10 +2,452 @@
...
+
+> There are many aspects of day-to-day analytical work that are almost absent from the conventional statistics literature and curriculum. And yet these activities account for a considerable share of the time and effort of data analysts and applied statisticians. The goal of this collection is to increase the visibility and adoption of modern data analytical workflows. We aim to facilitate the transfer of tools and frameworks between industry and academia, between software engineering and Stats/CS, and across different domains.
```
Comment (ismayc) [inline 12-thinking-with-data.Rmd]: Should probably write out "Stats/CS" as "Statistics/Computer Science"

File: 12-thinking-with-data.Rmd [hunk 12]
```diff
@@ -2,10 +2,452 @@
...
+  labs(y = "log10 price", x = "log10 size", title = "House prices in Seattle") +
+  geom_smooth(method = "lm", se = FALSE)
+```
+
+For homes of all 5 condition types, as the size of the house increases, the prices increases. This is what most would expect. However, the rate of increase of price with size is fastest for the homes with condition 3, 4, and 5 of 0.823, 0.836, and 1 respectively; these are the 3 most largest slopes out of the 5.
+
+
+### Inference for regression {#house-prices-inference-for-regression}
+
+Contributions via [pull request](http://happygitwithr.com/big-picture.html) to the [ModernDive GitHub repository](https://github.com/moderndive/moderndive_book) are welcome!
```
Comment (ismayc) [inline 12-thinking-with-data.Rmd]: Do we want this included here?

File: 12-thinking-with-data.Rmd [hunk 13]
```diff
@@ -2,10 +2,452 @@
...
+
+
+### Making predictions {#house-prices-making-predictions}
+
+Say your are a realtor and someone calls you asking you how much their home will sell for. They tell you that it's in condition = 5 and is sized 1900 square feet. What do you tell them? We first make this prediction visually in Figure \@ref(fig:house-price-interaction-3). The predicted `log10_price` of this house is marked with a black dot: it is where the two following lines intersect:
```
Comment (ismayc) [inline 12-thinking-with-data.Rmd]: "you're" instead of "your"

File: 07-multiple-regression.Rmd (hunk 10 above)
Comment (rudeboybert) [reply 07-multiple-regression.Rmd]: Yes

File: 06-regression.Rmd (hunk 6 above)
Comment (rudeboybert) [reply 06-regression.Rmd]: For a one time highlighting of 6 points, I feel like that's overkill. But I would not object you switched this out.

File: 12-thinking-with-data.Rmd (hunk 12 above)
Comment (rudeboybert) [reply 12-thinking-with-data.Rmd]: I don't have any material I can quickly cut and paste for this section. Let's comment this section out for now, especially since Ch11 isn't done yet.

File: 04-tidy.Rmd (hunk 2 above)
Comment (rudeboybert) [reply 04-tidy.Rmd]: I'll create a new issue so we can quickly merge this PR.
